crates/hotshot/src/lib.rs

```diff
@@ -386,6 +386,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
         self.inner.consensus.read().await.get_decided_leaf()
     }
 
+    /// Synchronously tries to return a copy of the last decided leaf,
```

Comment:
nit: When I see synchronously I think this is blocking maybe update the comment to say it won't block and fail if it can't get the lock without waiting